model/testresult/testresult.go

```diff
@@ -136,3 +136,17 @@ func TestResultsQuery(taskId, testId, status string, limit, sort, execution int)
 
 	return q
 }
+
+func FindByTaskIDAndExecutionGroupedByTestFile(taskID string, execution int) (map[string]*TestResult, error) {
```

Comment:
This needs to handle a task having multiple test results with the same name

Reply:
Yep, I'll fix that. The trigger also needs to be updated. I've asked Ian how he wants to handle this, so I'll hold off on making changes here just yet.